### DISCUSSION TITLE
"dist" folder in .gitignore

### DISCUSSION THREAD

**Discussion by danielbasso** [REditorSupport/vscode-R#1058] [General]
Silly question, but why the dist folder is in .gitignore?  I've launched the Extension Development Host and tried to `View()` some data frames, only to see a blank output. After some fiddling, I've realized that the .css files need to render the table were missing from the repository:

![image](https://user-images.githubusercontent.com/20959177/161434988-c7f56399-a8b3-42cb-b09c-e8596ea1518e.png)

After copying and pasting the files from the actual extension, everything came back to normal:

![image](https://user-images.githubusercontent.com/20959177/161435019-bfd0df25-9280-40ce-bc52-67477d3e199a.png)

No big deal, I'm just curious.

**Comment by renkun-ken** [comment]
You are right. The current build task does not populate `dist` folder created from the webpack config.

@ManuelHentschel Do you have any idea how we could fix this?

**Reply by ManuelHentschel** [reply]
Currently we have two sets of build tasks:
1. The ones you see in package.json and use during development. These are fast to compile and easy to debug.
2. The ones used for deployment, in the package.json under `"withWebpack"`. These use webpack and produce a smaller .vsix, but are significantly slower to compile and debug.

You can switch to the deployment build by running `node .github/scripts/enableWebpack.js` and back by resetting the git repo (so don't do this with uncommited changes). The dist folder is where the deployment tasks produce their output, so it is still in the `.gitignore` and is not intended to be used for other files.

Fyi this is where webpack is enabled in one of the Github actions: https://github.com/REditorSupport/vscode-R/blob/89e9633e280a98ae41005b658ef14c7f7611da6a/.github/workflows/main.yml#L28

Let me know if there are any issues with this setup!

**Reply by renkun-ken** [reply]
I'm curious if there is an easier or smarter way to do this? If the dist needs to be updated, then the build task should update it anyway.

**Reply by ManuelHentschel** [reply]
Just to be clear, what is the problem we would solve here? If I'm not mistaken, the only limitation caused by this setup is that a folder named `dist` cannot be used. If we used webpack permanently (or both build systems), the `dist` folder would still be populated by webpack and should not be used for other files.

**Reply by renkun-ken** [reply]
I'm no NodeJS expert, and I'm even not sure if I use `dist` in the proper way. I just need certain paths to those js/css files at runtime to be put in the WebViews.

**Reply by ManuelHentschel** [reply]
The files used by the plot and help webviews are currently in the 'html' subfolder. Those are included in the final vsix and can be accessed by their expected relative paths. Only the "main" ts/js files (including node modules) are optimised by webpack and put in the dist folder.

My personal recommendation would be to do the same for other webviews (if they don't need any node modules etc).
